build.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::process::Command;
+
+// this build script is called on en every `devtool build`,
+// embedding the FIRECRACKER_VERSION directly in the resulting binary
+fn main() {
+    let firecracker_version = Command::new("bash")
+        // the dirty flag reported by git describe --dirty does not suffice,
+        // since it is not affected by new, unstaged files;
+        // we check the output of git status to account for this case.
+        .args(&[
+            "-c",
+            "err=0; trap \"err=1\" ERR; git describe >/dev/null;\
+            echo -n \"$(git describe)\";\
+            [[ $(git status --porcelain | wc -l) -eq 0 ]] && : || echo '-dirty';\
+            test $err = 0",
+        ])
+        .output()
+        .ok()
+        .and_then(|output| {
+            if output.status.success() {
+                return Some(output.stdout);
+            }
+            None
+        })
+        .and_then(|version_bytes| String::from_utf8(version_bytes).ok())
+        .map(|version_string| version_string.trim_start_matches('v').to_string())
+        .unwrap_or_else(|| env!("CARGO_PKG_VERSION").to_string());
+
+    println!(
```

Comment:
We should add another line with
```
println!(
        "cargo:rustc-env=JAILER_VERSION={}",
        firecracker_version
    );
```
and then update [this](https://github.com/alindima/firecracker/blob/improve_version/src/jailer/src/main.rs#L19) accordingly.

Reply:
This opens another discussion. Doesn't it make sense then to have a single environment variable, to avoid duplicating this? Something like "REPO_VERSION" or just "FIRECRACKER_VERSION" for both jailer and firecracker (since the repo is named firecracker and the jailer is only a sub-crate). WDYT?

Reply:
You're right, since we're basing on a `git` command, the result will always be repo-level, so using a single env variable makes sense - `REPO_VERSION` and `FIRECRACKER_VERSION` are both fine.

Don't forget to consume the env var in the `jailer` binary as well.

Reply:
done in latest commit